juliet/src/rpc.rs

```diff
@@ -345,10 +394,13 @@ where
     /// Process all pending timeouts, setting and notifying `RequestError::TimedOut` on timeout.
     ///
     /// Returns the duration until the next timeout check needs to take place if timeouts are not
-    /// modified in the interim.
-    fn process_timeouts(&mut self, now: Instant) -> Instant {
+    /// modified in the interim, and the number of actual timeouts.
+    fn process_timeouts(&mut self, now: Instant) -> (Instant, usize) {
         let is_expired = |t: &Reverse<(Instant, IoId)>| t.0 .0 <= now;
 
+        // Track the number of actual timeouts hit.
+        let mut timed_out = 0;
```

Comment:
Why bother counting them if any non-zero value triggers the error?

Reply:
They are logged. In general, the info is free, so I thought it best to not just discard it in the library.